api/admin/controller/library_settings.py

```diff
@@ -294,19 +297,50 @@ def validate_geographic_areas(self, values):
                     elif len([x for x in us_search.query(state=state, returns=None) if x.county == city_or_county]):
                         locations["US"].append(value);
                     else:
-                        return UNKNOWN_LOCATION.detailed(_('Unable to locate "%(value)s".', value=value))
-
+                        # Flag this as needing to be checked with the registry
+                        flagged = True
                 elif value.isdigit():
                     # Is it a US zipcode?
                     info = us_search.by_zipcode(value)
                     if not info:
                         return UNKNOWN_LOCATION.detailed(_('"%(value)s" is not a valid U.S. zipcode.', value=value))
                     locations["US"].append(self.format_place(value, info.major_city, info.state));
                 else:
-                    return UNKNOWN_LOCATION.detailed(_('Unable to locate "%(value)s".', value=value))
+                    flagged = True
+
+            if flagged:
+                for nation in ["US", "CA"]:
+                    service_area_object = urllib.quote('{"%s": "%s"}' % (nation, value))
+                    registry_check = self.ask_registry(service_area_object)
+                    if registry_check and isinstance(registry_check, ProblemDetail):
+                        return registry_check
+                    elif registry_check:
+                        locations[nation].append(value)
+                        flagged = False
+                        # If the registry has established that this is a US location, don't bother also trying to find it in Canada
+                        break
+
+            if flagged:
+                return UNKNOWN_LOCATION.detailed(_('Unable to locate "%(value)s".', value=value))
 
         return json.dumps(locations)
 
     def format_place(self, zip, city, state_or_province):
         info = "%s, %s" % (city, state_or_province)
         return { zip: info }
+
+    def ask_registry(self, service_area_object):
+        # If the circulation manager doesn't know about this location, check whether the Library Registry does.
+        for registry in RemoteRegistry.for_protocol_and_goal(
+            self._db, ExternalIntegration.OPDS_REGISTRATION, ExternalIntegration.DISCOVERY_GOAL
+        ):
+            base_url = registry.integration.url + "/coverage?coverage="
+
+            response = HTTP.debuggable_get(base_url + service_area_object)
+            if not response.status_code == 200:
+                return REMOTE_INTEGRATION_FAILED.detailed(_("Unable to contact the registry at %(url)s.", url=registry.integration.url))
+
+            content = json.loads(response.content)
+            found_place = not (content.get("unknown") or content.get("ambiguous"))
+            if found_place:
+                return True
```

Comment:
Don't bail out immediately if one of the registries can't be contacted, but save the problem detail and at the end of the method, if there's a saved problem detail, return it rather than returning None.